packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,204 @@
+   * This does not modify the isChecked property.
+   *
+   * @default false
+   */
+  isIndeterminate?: boolean;
+  /**
+   * The name of the input field in a checkbox
+   * (Useful for form submission).
+   */
+  name?: string;
+  /**
+   * The value to be used in the checkbox input.
+   * This is the value that will be returned on form submission.
+   */
+  value?: string;
+  /**
+   * If `true`, the checkbox will be disabled
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * If `true`, the checkbox input is marked as required,
+   * and `required` attribute will be added
+   *
+   * @default false
+   */
+  isRequired?: boolean;
+  /**
+   * If `error`, the checkbox input is marked as invalid,
+   * and `invalid` attribute will be added
+   *
+   * @default undefined
+   */
+  validationState?: 'error' | undefined;
+};
+
+const Checkbox = ({
+  defaultChecked,
+  validationState,
+  isChecked,
+  isDisabled,
+  isIndeterminate,
+  isRequired,
+  name,
+  onChange,
+  value,
+  children,
+  helpText,
+  errorText,
+}: CheckboxProps): React.ReactElement => {
+  const groupProps = useCheckboxGroupContext();
+
+  // ban controlling checkbox manually while inside group
+  const hasDefaultChecked = !isUndefined(defaultChecked);
+  const hasIsChecked = !isUndefined(isChecked);
+  const hasOnChange = !isUndefined(onChange);
+  if ((hasDefaultChecked || hasIsChecked || hasOnChange) && !isEmpty(groupProps)) {
+    const props = [
+      hasDefaultChecked ? 'defaultChecked' : undefined,
+      hasIsChecked ? 'isChecked' : undefined,
+      hasOnChange ? 'onChange' : undefined,
+    ].filter(Boolean);
+
+    throw new Error(
+      `[Blade Checkbox]: Cannot set \`${props.join(
+        ',',
+      )}\` on <Checkbox /> when it's inside <CheckboxGroup />, Please set it on the <CheckboxGroup /> itself`,
+    );
+  }
+  // mandate value prop when using inside group
+  if (!value && !isEmpty(groupProps)) {
+    throw new Error(
+      `[Blade Checkbox]: <CheckboxGroup /> requires that you pass unique "value" prop to each <Checkbox />
+      <CheckboxGroup>
+        <Checkbox value="apple">Apple</Checkbox>
+        <Checkbox value="mango">Mango</Checkbox>
+      </CheckboxGroup>
+      `,
+    );
+  }
+  // If validationState is used while inside group
+  if (validationState && !isEmpty(groupProps)) {
+    throw new Error(
+      "[Blade Checkbox]: Can't use validationState of individual Checkboxes inside <CheckboxGroup />, please set validtionState on the <CheckboxGroup /> itself.",
+    );
+  }
+
+  const _validationState = validationState ?? groupProps.validationState;
+  const _hasError = _validationState === 'error';
+  const _isDisabled = isDisabled ?? groupProps.isDisabled;
+  const _name = name ?? groupProps.name;
+  const _isChecked = isChecked ?? groupProps.state?.isChecked(value!);
+
+  const showError = validationState === 'error' && errorText;
+  const showHelpText = !showError && helpText;
```

Comment:
we need to abstract all the helptext, errortext, label things into a diff component since even I'll be needing them for textinput. let's discuss this on Monday and start segregating them

Reply:
Not sure. the usecases are very dependent on what form component you are building. 

For example in here I don't want to render error if the individual checkbox's validation is set to error. PasswordField/TextField will have different requirements.

Reply:
Yes.